nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/ScanHBase.java

```diff
@@ -0,0 +1,564 @@
+            if (StringUtils.isBlank(tableName)) {
+                getLogger().error("Table Name is blank or null for {}, transferring to failure", new Object[] {flowFile});
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }
+
+            final String startRow = context.getProperty(START_ROW).evaluateAttributeExpressions(flowFile).getValue();
+            final String endRow = context.getProperty(END_ROW).evaluateAttributeExpressions(flowFile).getValue();
+
+            final String filterExpression = context.getProperty(FILTER_EXPRESSION).evaluateAttributeExpressions(flowFile).getValue();
+
+            //evaluate and validate time range min and max values. They both should be either empty or provided.
+            Long timerangeMin = null;
+            Long timerangeMax = null;
+
+            try{
+                timerangeMin = context.getProperty(TIME_RANGE_MIN).evaluateAttributeExpressions(flowFile).asLong();
+            }catch(Exception e){
+                getLogger().error("Time range min value is not a number ({}) for {}, transferring to failure",
+                        new Object[] {context.getProperty(TIME_RANGE_MIN).evaluateAttributeExpressions(flowFile).getValue(), flowFile});
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }
+            try{
+                timerangeMax = context.getProperty(TIME_RANGE_MAX).evaluateAttributeExpressions(flowFile).asLong();
+            }catch(Exception e){
+                getLogger().error("Time range max value is not a number ({}) for {}, transferring to failure",
+                        new Object[] {context.getProperty(TIME_RANGE_MAX).evaluateAttributeExpressions(flowFile).getValue(), flowFile});
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }
+            if (timerangeMin == null && timerangeMax != null) {
+                getLogger().error("Time range min value cannot be blank when max value provided for {}, transferring to failure", new Object[] {flowFile});
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }else if (timerangeMin != null && timerangeMax == null) {
+                getLogger().error("Time range max value cannot be blank when min value provided for {}, transferring to failure", new Object[] {flowFile});
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }
+
+            final Integer limitRows = context.getProperty(LIMIT_ROWS).evaluateAttributeExpressions(flowFile).asInteger();
+
+            final Boolean isReversed = context.getProperty(REVERSED_SCAN).asBoolean();
+
+            final Integer bulkSize = context.getProperty(BULK_SIZE).evaluateAttributeExpressions(flowFile).asInteger();
```

Comment:
What happens if limitRows or bulkSize reference an attribute that doesn't exist?

I think these will end up as null and I can't tell if the rest of the code protects against that case.

Reply:
limitRows handled: HBase_1_1_2_ClientService.scan line 443
bulkSize missed :-O   (fixing)